### ISSUE TITLE
Latest version cannot find "wkhtmltox.dll"

### ISSUE THREAD

**Issue by appstractmeyn** [tuespetre/TuesPechkin#79]
I notice the latest version creates the wkhtmltox.dll in a sub folder 
\wkhtml-something\0.12.2.1\wkhtmltox.dll
The older version creates the wkhtmltox.dll  it in a folder above.
So I have it in \wkhtml-something\wkhtmltox.dll

The problem with the latest version is, that it works, when I run it locally in Visual Studio, the error I get on the server is that "wkhtmltox.dll cannot be found".
Reverting to the older version of TuesPechkin, makes this problem disappear.

Unfortunately, I cannot use the older version, because my print.css does not get used unless I use the latest version.

So basically I don't have a version that works completely.

**Comment by tuespetre** [comment]
The error message you are describing, if I remember right, is what happens when the Visual Studio C++ Runtime is not installed. Please ensure that this is installed on your server as well. You can also deploy the mingw version yourself but checking VC++ is easier.

**Comment by richardlawley** [comment]
I've just arrived here looking for the solution to this as well, and the cause was indeed the VC runtimes.  Perhaps the exception could be trapped and explained to aid people in future with this problem.  Here's the relevant part of the exception if this helps:

```
[DllNotFoundException: Unable to load DLL 'wkhtmltox.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)]
   TuesPechkin.ThreadSafeConverter.Invoke(FuncShim`1 delegate) +256
   [Removed]
```

I was running this inside an MVC application under IIS.

**Comment by neil-119** [comment]
Same issue here. I get a DllNotFoundException, even though I have set permissions for read access on IIS and installed VC 2013 redist. In my last effort, I copied the DLL from the x64 nuget package and put the file in the root of my project, and tried to use StaticDeployment with the path of the DLL. It still throws DllNotFoundException, but File.Exists() says the DLL does exist at the path.

Anyway, for now I've resorted back to the old version of TuesPechkin (the one with Factory.Create(), but I'm not sure if it's safe to use with IIS). Just wanted to mention, thanks for this great project --- your work does not go unappreciated!

**Comment by AncientBeing** [comment]
Use on IIS remains complicated, locally it runs smoothly with the TuesPechkin.Wkhtmltox.Win32, on my remote host I require  TuesPechkin.Wkhtmltox.Win64, any idea how to achieve this in an ASP.NET MVC application? Should i create two converters and use the one or other depended on the web.config?

Does anybody know if Visual C++ 2010 (not 2013) Runtime is sufficient?

IConverter converter32 =
    new ThreadSafeConverter(
        new RemotingToolset<PdfToolset>(
            new Win32EmbeddedDeployment(
                new TempFolderDeployment())));

IConverter converter64 =
    new ThreadSafeConverter(
        new RemotingToolset<PdfToolset>(
            new Win64EmbeddedDeployment(
                new TempFolderDeployment())));

**Comment by richardlawley** [comment]
Switch Visual Studio to using 64-bit IIS and your life will be easier (Tools > Options, Projects and Solutions > Web Projects > "Use the 64 bit version of IIS Express for web sites and projects).

You need Visual C++ 2013 64 bit runtime.

**Comment by AncientBeing** [comment]
Ah, thanks, this sounds quite good. Will have a try asap. My Hoster seems only to have C++ 2010 running however, I hope he is willing to update. So many issues with hosted .NET :-(

**Comment by tuespetre** [comment]
@appstractmeyn let me know or reopen if you still have this issue.

**Comment by neil-119** [comment]
No matter what I do, I can't get the new version to work. Same error as OP.
Should I open a new bug report?

**Comment by tuespetre** [comment]
You can try the MingW flavor of wkhtmltopdf instead, it has no VC++ dependency. 

http://wkhtmltopdf.org/downloads.html

Future versions of the TuesPechkin.Wkhtmltox.WinXX packages will use the MingW versions of Wkhtmltopdf as they are released, to avoid this confusion

**Comment by AncientBeing** [comment]
I got TuesPechkin with ThreadSafe/Embedded/32-bit and 64-bit versions running locally with iisexpress switching from 32/64 bit mode like richardlawley mentioned. For 64-bit i had to install Visual C++ 2013 64 bit runtime (previously i had only 32 bit installed on my machine).

However, when "Publishing" to my local IIS (7.5, Windows 7 64bit) i get a ASP.NET COMPILATION ERROR, there still seems to be an issue with 64 bit stuff.

Update:
Worked after adding 
    <AspnetCompilerPath>$(windir)\Microsoft.NET\Framework64\v4.0.30319</AspnetCompilerPath>
to the Configuration and disabling 32bit Mode on IIS 7.5

Currently i'dont dare to deploy however :-)

**Comment by AncientBeing** [comment]
Can someone tell me how to use MingW wkhtmltopdf with the Converter?

IConverter converter32 =
new ThreadSafeConverter(
new RemotingToolset(
new TempFolderDeployment()));

**Comment by canbaran14** [comment]
Hello, I have builded my project as a dll and published to UiPath Tool which has .net framework but It says "Unable to find assembly 'TuesPechkin.Wkhtmltox.Win32, Version=0.12.2.1" Anyone can help me? btw My project works on Visiual studio
